### PR TITLE
Added SetSpecial command 

### PR DESCRIPTION
**Pull request by dim**
Proposal for issue #66

### CODE REVIEW

**Comment by vmihailenco** [comment]
We could add some logic to automatically use EX or PX as needed. And use more descriptive name :)

As for the NX|XX we can have SetNX(key, value, expiration) and SetXX(key, value, expiration). Since `SetNX` is already used we should probably release it as v3.

**Comment by vmihailenco** [comment]
@bronze1man @dim any thoughts?

**Comment by dim** [comment]
As the redis documentation suggests:

> Note: Since the SET command options can replace SETNX, SETEX, PSETEX, it is possible that in future versions of Redis these three commands will be deprecated and finally removed.

That's the primary reason why I went for a `SetSpecial`. I don't really have a name preference though.

**Comment by bronze1man** [comment]
I think it is a good idea to add SetSpecial,and rename to set in v4.

**Comment by vmihailenco** [comment]
I've sent https://github.com/go-redis/redis/pull/69 to demonstrate what I mean. So far I like it much more than `SetSpecial` approach.

>That's the primary reason why I went for a SetSpecial

When Redis will remove support for `SETNX` go-redis will use `SET key value NX` as a replacement. No API changes will be needed.

**Comment by bronze1man** [comment]
I think #69 is better.#69 is simpler then #67.
But I think #69 need API changes.

**Comment by vmihailenco** [comment]
Yeah, it is supposed to be released in new version. I don't mind to merge this PR into v2 branch if there is interest.

File: commands.go [hunk 1]
```diff
@@ -429,6 +429,29 @@ func (c *Client) Set(key, value string) *StatusCmd {
 	return cmd
 }
 
+type SetOpt struct {
+	EX, PX time.Duration
+	NX, XX bool
+}
+
+func (c *Client) SetSpecial(key, value string, opt SetOpt) *StatusCmd {
```
Comment (vmihailenco) [inline commands.go]: I would name it `SetExt` or `SetExtended` if we are going to add it into v2 branch.

**Comment by vmihailenco** [comment]
Closing in favor of https://github.com/go-redis/redis/pull/69

**Pull request closed without merging**